app/validators/case_details/answers_validator.rb

```diff
@@ -1,7 +1,88 @@
 module CaseDetails
-  class AnswersValidator < ActiveModel::Validator
-    def validate(record)
-      record.errors.add :base, :incomplete_records unless record.complete?
+  class AnswersValidator
+    def initialize(record)
+      @record = record
+    end
+
+    attr_reader :record
+    alias kase record
+
+    delegate :errors, :crime_application, to: :kase
+
+    # Adds the error to the first step name a user would need to go to
+    # fix the issue.
+
+    def validate # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+      errors.add(:has_case_concluded, :blank) unless case_concluded_complete?
+
+      unless not_means_tested?
+        errors.add(:is_preorder_work_claimed, :blank) unless preorder_work_complete?
+        errors.add(:is_client_remanded, :blank) unless client_remanded_complete?
+      end
+
+      errors.add(:charges, :blank) unless what_charges_complete?
+      errors.add(:charges_summary, :incomplete_records) unless all_charges_complete?
+      errors.add(:has_codefendants, :blank) unless has_codefendants_complete?
+      errors.add(:codefendants_summary, :incomplete_records) unless all_codefendants_complete?
+      errors.add(:hearing_details, :blank) unless hearing_details_complete?
+      errors.add(:first_court_hearing, :blank) unless first_court_hearing_complete?
+
+      errors.add :base, :incomplete_records unless errors.empty?
+    end
+
+    delegate :not_means_tested?, to: :crime_application
+
+    def case_concluded_complete?
+      return false if kase.has_case_concluded.blank?
+      return true unless kase.has_case_concluded == 'yes'
```

Comment:
This condition is a little different to the `preorder_work_complete` and `client_remanded_complete` below. The logic is correct but perhaps they should be the same for consistency?

```suggestion
      return true if kase.has_case_concluded == 'no'
```

Reply:
Thanks because the pre-order is dependent on the case concluded question and can be blank. (Where as the others can never be blank and complete.)